consensus/spos/bls/subroundSignature.go

```diff
@@ -302,3 +310,71 @@ func (sr *subroundSignature) remainingTime() time.Duration {
 
 	return remainigTime
 }
+
+func (sr *subroundSignature) doSignatureJobForManagedKeys() bool {
+	isAllInOneLeader := sr.IsMultiKeyLeaderInCurrentRound()
+
+	managedKeys := sr.GetManagedKeysByCurrentNode()
+	numMultiKeysSignaturesSent := 0
+	for pk, sk := range managedKeys {
+		if !sr.IsNodeInConsensusGroup(pk) {
+			continue
+		}
+
+		if sr.IsJobDone(pk, sr.Current()) {
+			continue
+		}
+
+		signatureShare, err := sr.MultiSigner().CreateAndAddSignatureShareForKey(sr.GetData(), sk, []byte(pk))
+		if err != nil {
+			log.Debug("doSignatureJob.CreateAndAddSignatureShareForKey", "error", err.Error())
+			return false
+		}
+
+		if !isAllInOneLeader {
+			cnsMsg := consensus.NewConsensusMessage(
```

Comment:
this part looks very similar to L73-L118
can we extract some common functionality and use in both places?

Reply:
refactored and eliminated duplicated code